src/general/GameWorld.cs

```diff
@@ -310,11 +319,53 @@ public Species GetSpecies(uint id)
         return worldSpecies[id];
     }
 
+    public void LogWorldEvent(LocalizedString description, bool highlight = false, string iconPath = null)
+    {
+        // Enforce limit
+        if (worldTimeline.Count >= Constants.TIMELINE_HISTORY_RANGE)
+        {
+            var oldestKey = worldTimelineKeys.Dequeue();
+            worldTimeline.Remove(oldestKey);
+        }
+
+        if (!worldTimeline.ContainsKey(TotalPassedTime))
+            worldTimeline.Add(TotalPassedTime, new List<WorldEventDescription>());
+
+        // Event already logged in timeline
+        if (worldTimeline[TotalPassedTime].Any(entry => entry.EventDescription.ToString() == description.ToString()))
+            return;
+
+        worldTimeline[TotalPassedTime].Add(new WorldEventDescription
+        {
+            EventDescription = description,
+            IconPath = iconPath,
+            Highlighted = highlight,
+        });
+
+        worldTimelineKeys.Enqueue(TotalPassedTime);
+    }
+
+    public IReadOnlyDictionary<double, List<WorldEventDescription>> GetTimeline()
```

Comment:
Instead of a getter, make a read only property to read the timeline through it as this type.